python/spglib/spglib.py

```diff
@@ -97,6 +103,101 @@ class SpglibError:
 spglib_error = SpglibError()
 
 
+@dataclasses.dataclass(eq=True, frozen=True)
+class SpaceGroupType(Mapping[str, Any]):
+    """Space group type information.
+
+    Dictionary keys are as follows:
+
+    - number : int
+        International space group number
+    - international_short : str
+        International short symbol.
+    - international_full : str
+        International full symbol.
+    - international : str
+        International symbol.
+    - schoenflies : str
+        Schoenflies symbol.
+    - hall_number : int
+        Hall symbol ID number.
+    - hall_symbol : str
+        Hall symbol.
+    - choice : str
+        Centring, origin, basis vector setting.
+    - pointgroup_international : str
+        International symbol of crystallographic point group.
+    - pointgroup_schoenflies : str
+        Schoenflies symbol of crystallographic point group.
+    - arithmetic_crystal_class_number : int
+        Arithmetic crystal class number
+    - arithmetic_crystal_class_symbol : str
+        Arithmetic crystal class symbol.
+    """
+
+    number: int
+    international_short: str
+    international_full: str
+    international: str
+    schoenflies: str
+    hall_number: int
+    hall_symbol: str
+    choice: str
+    pointgroup_international: str
+    pointgroup_schoenflies: str
+    arithmetic_crystal_class_number: int
+    arithmetic_crystal_class_symbol: str
+
+    def __getitem__(self, key: str) -> Any:
+        """Return the value of the key."""
+        warnings.warn(
+            f"dict interface ({self.__class__.__name__}['{key}']) is deprecated."
+            "Use attribute interface ({self.__class__.__name__}.{key}) instead",
+            DeprecationWarning,
+        )
+        return dataclasses.asdict(self)[key]
+
+    def __len__(self) -> int:
+        """Return the number of fields."""
+        return len(dataclasses.fields(self))
+
+    def __iter__(self) -> Iterator[str]:
+        """Return an iterator over the keys."""
+        return iter(dataclasses.asdict(self))
+
+
+@dataclasses.dataclass(eq=True, frozen=True)
+class MagneticSpaceGroupType(Mapping[str, Any]):
+    """Magnetic space group type information.
+
+    See :ref:`api_get_magnetic_spacegroup_type` for attributes.
```

Comment:
Better to include the documentation in the type. It will pop-up on most IDE's when they hover over it.